src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -138,3 +140,57 @@ def file_get_contents(self, path, options=None):
                 f.seek(options['offset'])
             data = binascii.b2a_base64(f.read(options.get('maxlen'))).decode().strip()
         return data
+
+
+class GetAndWatchEventSource(EventSource):
+
+    def run(self):
+        if ':' in self.arg:
+            path, lines = self.arg.rsplit(':', 1)
+            lines = int(lines)
+        else:
+            path = self.arg
+            lines = 3
+        if not os.path.exists(path):
+            # FIXME: Error?
+            return
+
+        bufsize = 8192
+        fsize = os.stat(path).st_size
+        if fsize < bufsize:
+            bufsize = fsize
+        i = 0
+        with open(path) as f:
+            data = []
+            while True:
```

Comment:
More efficient implementation is possible if you are really into this :)
For example, this is how it's done in GNU tail: http://git.savannah.gnu.org/cgit/coreutils.git/tree/src/tail.c#n502